### PR TITLE
A bunch of improvements

### LINKED ISSUE #8: Support PLEX_MEDIA_SERVER_* vars
Template for `/etc/default/plexmediaserver`, with attributes for config values. Most notably `PLEX_MEDIA_SERVER_APPLICATION_SUPPORT_DIR`.

### LINKED ISSUE #10: use dpkg_package with the source attrib, apt_package doesn't work properly
Hey man,

I gave this cookbook a whirl tonight in a vagrant env, and it blew up on me as follows.

apt_package doesn't seem to support the style of invocation that is in use here - changing from package to dpkg_package makes it Just Work.

best,

--e

> 8>8>8>8>8>8

==> default:
==> default: ================================================================================
==> default: Error executing action `install` on resource 'apt_package[plexmediaserver]'
==> default: ================================================================================
==> default:
==> default:
==> default: Chef::Exceptions::Package
==> default: -------------------------
==> default: apt package provider cannot handle source attribute. Use dpkg provider instead
==> default:
==> default: Resource Declaration:
==> default: ---------------------
==> default: # In /tmp/vagrant-chef-5/chef-solo-1/cookbooks/plexapp/recipes/server.rb
==> default:
==> default:  31: package 'plexmediaserver' do
==> default:  32:   source cached_package_file
==> default:  33: end
==> default:  34:
==> default:

### PR DESCRIPTION
**Pull request by odcinek**
- templated config file
- smarter service restart
- do not reinstall package if version already installed

Closes https://github.com/jtimberman/plexapp-cookbook/pull/6, closes https://github.com/jtimberman/plexapp-cookbook/issues/8, closes https://github.com/jtimberman/plexapp-cookbook/issues/10.

### CODE REVIEW

File: recipes/server.rb [hunk 1]
```diff
@@ -28,8 +28,23 @@
...
+  notifies :restart, 'service[plexmediaserver]'
+  options '--force-confold'
+  action :install
+  version node['plexapp']['version']
+end if platform_family?('debian')
+
 package 'plexmediaserver' do
   source cached_package_file
+  notifies :restart, 'service[plexmediaserver]'
+end unless platform_family?('debian')
```
Comment (mburns) [inline recipes/server.rb]: Why have two (very similar) resources when we should be able to make a single one work?

```ruby
package 'plexmediaserver' do
  source cached_package_file
  notifies :restart, 'service[plexmediaserver]'
  options '--force-confold' if platform_family?('debian')
  version node['plexapp']['version'] if platform_family?('debian')
end
```

**Comment by tfitch** [comment]
This PR resolves Issue #10 I see with the current master and Ubuntu 15.04.

File: recipes/server.rb (hunk 1 above)
Comment (odcinek) [inline recipes/server.rb]: See https://github.com/jtimberman/plexapp-cookbook/issues/10, and https://github.com/chef/chef/issues/1763.

File: metadata.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{
```
Comment (josephholsten) [inline metadata.json]: pretty sure we don't want the metadata.json

**Comment by josephholsten** [comment]
@jtimberman I can confirm this is working well on debian/ubuntu. I'd cut the `metadata.json` file, but that should be an easy patch if you'd like.

**Comment by cheeseplus** [comment]
+1

**Comment by jtimberman** [comment]
I've added @cheeseplus as a collaborator here and on supermarket.

**Pull request merged**